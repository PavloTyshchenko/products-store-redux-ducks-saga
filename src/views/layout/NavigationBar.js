import React from 'react';
import { Link } from 'react-router-dom';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { LinkContainer } from "react-router-bootstrap";

const NavigationBar = () => {

    return (
        <Navbar bg="primary" variant="dark" className="mb-3">
            <LinkContainer to="/">
                <Navbar.Brand>Navbar</Navbar.Brand>
            </LinkContainer>
            <Nav className="mr-auto">
                <Nav.Link as={Link} to='/'>Home</Nav.Link>
                <Nav.Link as={Link} to='/products/All'>Products</Nav.Link>
                <Nav.Link as={Link} to='/about'>About</Nav.Link>
            </Nav>

        </Navbar>
    );
};

export default NavigationBar;
