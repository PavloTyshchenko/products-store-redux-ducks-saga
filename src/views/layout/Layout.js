import React from 'react'
import { Container } from 'react-bootstrap';

import NavigationBar from './NavigationBar';

const Layout = props => {
    return (
        <Container data-test="container">
            <NavigationBar data-test="navBar"/>
            {props.children}
        </Container>
    );
};

export default Layout;
