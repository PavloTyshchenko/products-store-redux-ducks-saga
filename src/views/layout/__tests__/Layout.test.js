import React from 'react';
import { shallow } from 'enzyme';

import Layout from '../Layout';
import NavigationBar from '../NavigationBar';


describe('Layout Component', () => {

    it('Should render without errors', () => {
        const component = shallow(<Layout />);
        expect(component.length).toBe(1);
    });

    it('Should render a Container', () => {
        const component = shallow(<Layout />);
        const container = component.find(`[data-test='container']`);
        expect(container.length).toBe(1);
    });

    it('Should render a NavigationBar', () => {
        const component = shallow(<Layout />);
        const navBar = component.find(`[data-test='navBar']`);
        expect(navBar.length).toBe(1);
    });

    test('Should match snapshot', () => {
        const component = shallow(<Layout />);
        expect(component.debug()).toMatchSnapshot();
    });
});