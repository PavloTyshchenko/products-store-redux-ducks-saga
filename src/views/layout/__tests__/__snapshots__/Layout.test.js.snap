// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Should match snapshot 1`] = `
"<Container data-test=\\"container\\" fluid={false}>
  <NavigationBar data-test=\\"navBar\\" />
</Container>"
`;
